src/include/duckdb/common/adbc/adbc.hpp

```diff
@@ -10,10 +10,255 @@
+			}
+			delete array;
+		}
+	}
+	ArrowArray *get() const {
+		return array;
+	}
+	ArrowArray &operator*() const {
+		return *array;
+	}
+	ArrowArray *operator->() const {
+		return array;
+	}
+	ArrowArrayWrapper(const ArrowArrayWrapper &) = delete;
+	ArrowArrayWrapper &operator=(const ArrowArrayWrapper &) = delete;
+	ArrowArrayWrapper(ArrowArrayWrapper &&other) noexcept : array(other.array) {
+		other.array = nullptr;
+	}
+	ArrowArrayWrapper &operator=(ArrowArrayWrapper &&other) noexcept {
+		if (this != &other) {
+			if (array && array->release) {
+				array->release(array);
+			}
+			delete array;
+			array = other.array;
+			other.array = nullptr;
+		}
+		return *this;
+	}
+
+private:
+	ArrowArray *array;
+};
+
+class AppenderWrapper {
+public:
+	AppenderWrapper(duckdb_connection conn, const char *schema, const char *table) : appender(nullptr) {
+		if (duckdb_appender_create(conn, schema, table, &appender) != DuckDBSuccess) {
+			appender = nullptr;
+		}
+	}
+	~AppenderWrapper() {
+		if (appender) {
+			duckdb_appender_destroy(&appender);
+		}
+	}
+
+	duckdb_appender get() const {
+		return appender;
+	}
+	explicit operator duckdb_appender() const {
+		return appender;
+	}
+	bool valid() const {
+		return appender != nullptr;
+	}
+	AppenderWrapper(const AppenderWrapper &) = delete;
+	AppenderWrapper &operator=(const AppenderWrapper &) = delete;
+	AppenderWrapper(AppenderWrapper &&other) noexcept : appender(other.appender) {
+		other.appender = nullptr;
+	}
+	AppenderWrapper &operator=(AppenderWrapper &&other) noexcept {
+		if (this != &other) {
+			if (appender) {
+				duckdb_appender_destroy(&appender);
+			}
+			appender = other.appender;
+			other.appender = nullptr;
+		}
+		return *this;
+	}
+
+private:
+	duckdb_appender appender;
+};
+
+class DataChunkWrapper {
+public:
+	DataChunkWrapper() : chunk(nullptr) {
+	}
+
+	~DataChunkWrapper() {
+		if (chunk) {
+			duckdb_destroy_data_chunk(&chunk);
+		}
+	}
+	explicit operator duckdb_data_chunk() const {
+		return chunk;
+	}
+
+	DataChunkWrapper(const DataChunkWrapper &) = delete;
+	DataChunkWrapper &operator=(const DataChunkWrapper &) = delete;
+	DataChunkWrapper(DataChunkWrapper &&other) noexcept : chunk(other.chunk) {
+		other.chunk = nullptr;
+	}
+	DataChunkWrapper &operator=(DataChunkWrapper &&other) noexcept {
+		if (this != &other) {
+			if (chunk) {
+				duckdb_destroy_data_chunk(&chunk);
+			}
+			chunk = other.chunk;
+			other.chunk = nullptr;
+		}
+		return *this;
+	}
+
+	duckdb_data_chunk chunk;
+};
+
+class ConvertedSchemaWrapper {
+public:
+	ConvertedSchemaWrapper() : schema(nullptr) {
+	}
+	~ConvertedSchemaWrapper() {
+		if (schema) {
+			duckdb_destroy_arrow_converted_schema(&schema);
+		}
+	}
+	duckdb_arrow_converted_schema *get_ptr() {
+		return &schema;
+	}
+
+	explicit operator duckdb_arrow_converted_schema() const {
+		return schema;
+	}
+	duckdb_arrow_converted_schema get() const {
+		return schema;
+	}
+	void set(duckdb_arrow_converted_schema s) {
+		schema = s;
```

Comment:
This should check whether `schema` isn't `nullptr` right? otherwise we're leaking the old one?